packages/services/api/src/modules/schema/providers/registry-checks.ts

```diff
@@ -184,46 +177,31 @@ export class RegistryChecks {
       ),
     );
 
-    try {
-      const policyResult = await this.policy.checkPolicy(sdl.raw, modifiedSdl, selector);
-      const warnings =
-        policyResult?.warnings?.map<SchemaCheckWarning>(record => ({
-          message: record.message,
-          source: record.ruleId ? `policy-${record.ruleId}` : 'policy',
-          column: record.column,
-          line: record.line,
-        })) ?? [];
-
-      if (policyResult === null) {
-        return {
-          status: 'skipped',
-        } satisfies CheckResult;
-      }
-
-      if (policyResult.success) {
-        return {
-          status: 'completed',
-          result: {
-            warnings,
-          },
-        } satisfies CheckResult;
-      }
+    const policyResult = await this.policy.checkPolicy(sdl.raw, modifiedSdl, selector);
+    const warnings = policyResult?.warnings?.map<SchemaCheckWarning>(toSchemaCheckWarning) ?? [];
 
+    if (policyResult === null) {
       return {
-        status: 'failed',
-        reason: {
-          errors: policyResult.errors,
-          warnings,
-        },
+        status: 'skipped',
       } satisfies CheckResult;
-    } catch (e) {
+    }
+
+    if (policyResult.success) {
       return {
-        status: 'failed',
-        reason: {
-          errors: [{ message: (e as Error).message }],
+        status: 'completed',
+        result: {
+          warnings,
         },
       } satisfies CheckResult;
     }
+
+    return {
+      status: 'failed',
+      reason: {
+        errors: policyResult.errors.map(toSchemaCheckWarning),
```

Comment:
We now transform both the warnings and errors in order to be more consistent. This is helpful for doing https://github.com/kamilkisiela/graphql-hive/issues/333